Symantec_docs_code.md

```diff
@@ -0,0 +1,112 @@
+
+The biggest overall win, though, came from the communication between dev and doc that sharing a workflow created. There simply couldn't be such a thing as an undocumented feature. And in some cases writer questions about what was going on with an error message, for example, produced better error handling generally. I was also able to catch some potential security issues, for example when a set of enums was initially called from the implementation code, potentially exposing black box functionality. The enums had to be documented for the web services, and so the team reluctantly realized that we'd have to violate DRY principles to expose the enums appropriately.
+
+## Cautionary tales
+
+Doc management was continually concerned that putting a writer into the code -- writing code comments and code samples -- would set developer expectations of the writers too high. In other words, it was seen as an un-extensible model. But the web services project was pretty well contained (I continued to work on other aspects of the server docs), and in fact there was quite a lot of code savvy on our docs team as a whole that management had previously been unaware of.
+
+The biggest difficulty came when we moved from SOAP to REST. Code comments don't work as source content for REST web services documentation, and it took us a while to figure out what to use instead. The tooling that we eventually settled on, spring-restdocs, required even tighter integration of dev and doc effort because the main doc files are written by hand, but the example requests and responses are generated from a documentation method that's added to the mock tests. So dev has to write the test method, either dev or the writer creates the main source files that include the examples, and then any reference information must be maintained manually outside the code, but within the code framework. These issues were still being addressed when I left the company.
+
+Generally the kind of approach that we took is at least initially writer-intensive, and therefore not extensible without careful planning and attention to the nuances of load balancing. For the first set of web services that we shipped, I worked on the project full-time. For updates the following year, we had enough of a system in place that I could work on the updates about half time. But juggling a developer workflow and a separate documentation workflow was quite a challenge, especially when I came back to the developer workflow after nearly a year of work in the separate doc flow. Developer tools and configurations had changed, and I had to learn them all over again. I don't necessarily recommend an all-or-nothing approach (all developer tools all the time or else none), but it's important for everyone -- doc teams and dev teams -- to understand the difficulties of jumping into and out of different workflows and toolchains. 
+
```

Comment:
I specifically wondered about the context switch so I'm glad you mention it. That is tough.